test/src/tests-serverModel.ts

```diff
@@ -0,0 +1,1675 @@
+            const uploadObject = {
+                ConsentState: {
+                    getGDPRConsentState: () => {
+                        return {
+                            'test-gdpr-purpose': {
+                                Consented: true,
+                                Timestamp: 12345,
+                                ConsentDocument: 'gdpr-doc',
+                                Location: 'test-gdpr-location',
+                                HardwareId: 'test-gdpr-hardware-id',
+                            },
+                        };
+                    },
+                    getCCPAConsentState: () => {
+                        return {
+                            Consented: false,
+                            Timestamp: 67890,
+                            ConsentDocument: 'ccpa-doc',
+                            Location: 'test-ccpa-location',
+                            HardwareId: 'test-ccpa-hardware-id',
+                        };
+                    },
+                } as SDKConsentState,
+            } as IUploadObject;
+
+            const actualDTO = ServerModel.convertEventToDTO(uploadObject);
+
+            expect(actualDTO.con, 'event.ConsentState (con)').to.eql({
+                gdpr: {
+                    'test-gdpr-purpose': {
+                        c: true,
+                        d: 'gdpr-doc',
+                        h: 'test-gdpr-hardware-id',
+                        l: 'test-gdpr-location',
+                        ts: 12345,
+                    },
+                },
+                ccpa: {
+                    data_sale_opt_out: {
+                        c: false,
+                        d: 'ccpa-doc',
+                        h: 'test-ccpa-hardware-id',
+                        l: 'test-ccpa-location',
+                        ts: 67890,
+                    },
+                },
+            });
+        });
+
+        it('should add AST data to DTO', () => {
+            const uploadObject = ({
+                EventDataType: Types.MessageType.AppStateTransition,
+                IsFirstRun: true,
+                LaunchReferral: 'https://mparticle.com/test-referral',
+                EventAttributes: {
+                    foo: 'bar', // TODO: test will nullify these
+                },
+            } as unknown) as IUploadObject;
+
+            const actualDTO = ServerModel.convertEventToDTO(uploadObject);
+
+            expect(actualDTO.fr, 'event.IsFirstRun (fr)').to.equal(true);
+
+            // TODO: What is IU?
+            expect(actualDTO.iu, '(iu)').to.equal(false);
+            expect(
+                actualDTO.at,
+                'event.ApplicationTransitionType.AppInit (at)'
+            ).to.equal(Types.ApplicationTransitionType.AppInit);
+            expect(actualDTO.lr, 'event.LaunchReferral (lr)').to.equal(
+                'https://mparticle.com/test-referral'
+            );
+
+            // TODO: Why do we nullify this?
```

Comment:
I think because there can't be any event attributes on an AST (not sure why, perhaps a server required thing).  It's unlikely that someone would ever log an AST on their own intentionally, but it's possible, via `logBaseEvent` where you can set the `MessageType` to be an AST

Reply:
Cool. I'll add a comment for posterity.